app/repository/sql/pipeline.sql

```diff
@@ -0,0 +1,178 @@
+    having count(*) = 1
+  ) non_duplicates
+  left join slug
+  on non_duplicates.family_import_id = slug.family_import_id
+  order by slug.family_import_id desc, slug.created desc, slug.ctid desc
+  ), most_recent_family_slugs as (
+  SELECT
+  deduplicated_family_slugs.family_import_id as "family_import_id",
+  deduplicated_family_slugs.created as "created",
+  deduplicated_family_slugs.name as "name"
+  FROM deduplicated_family_slugs
+  UNION ALL
+  SELECT
+  unique_family_slugs.family_import_id as "family_import_id",
+  unique_family_slugs.created as "created",
+  unique_family_slugs.name as "name"
+  FROM unique_family_slugs
+  order by family_import_id desc, created desc
+  ), deduplicated_doc_slugs as (
+  SELECT
+  distinct ON (slug.family_document_import_id)
+  slug.family_document_import_id,
+  slug.created,
+  slug.name
+  FROM (
+    SELECT
+    slug.family_document_import_id as "family_document_import_id",
+    count(*) as count
+    FROM slug
+    WHERE slug.family_document_import_id is not null
+    group by slug.family_document_import_id
+    having count(*) >  1
+  ) duplicates
+  left join slug
+  on duplicates.family_document_import_id = slug.family_document_import_id
+  order by
+  slug.family_document_import_id desc, slug.created desc, slug.ctid desc
+),
+unique_doc_slugs as (
+  SELECT
+  distinct ON (slug.family_document_import_id)
+  slug.family_document_import_id,
+  slug.created,
+  slug.name
+  FROM (
+    SELECT
+    slug.family_document_import_id as "family_document_import_id",
+    count(*) as count
+    FROM slug
+    WHERE slug.family_document_import_id is not null
+    group by slug.family_document_import_id
+    having count(*) = 1
+  ) non_duplicates
+  left join slug
+  on non_duplicates.family_document_import_id = slug.family_document_import_id
+  order by
+  slug.family_document_import_id desc, slug.created desc, slug.ctid desc
+  ), most_recent_doc_slugs as (
+  SELECT
+  deduplicated_doc_slugs.family_document_import_id
+  as "family_document_import_id",
+  deduplicated_doc_slugs.created,
+  deduplicated_doc_slugs.name
+  FROM deduplicated_doc_slugs
+  UNION ALL
+  SELECT
+  unique_doc_slugs.family_document_import_id as "family_document_import_id",
+  unique_doc_slugs.created,
+  unique_doc_slugs.name
+  FROM unique_doc_slugs
+  order by family_document_import_id desc, created desc
+  ), event_dates as (
+  SELECT
+  family_event.family_import_id as family_import_id,
+  min(CASE
+  WHEN family_event.event_type_name='Passed/Approved' then
+    family_event.date::date
+  ELSE family_event.date::date
+end) published_date,
+max(family_event.date::date) last_changed
+FROM family_event
+group by family_import_id
+)
+
+SELECT
+  f.title as "family_title",
+  p.title as "physical_document_title",
+  f.description as "family_description",
+  CASE
+    WHEN f.family_category in ('UNFCCC', 'MCF') THEN UPPER(f.family_category::TEXT)
+    ELSE INITCAP(f.family_category::TEXT)
+  END "family_category",
```

Comment:
Will this need updating every time an upper cased corpus is added? You could do something like the following instead:
```
WHEN f.family_category = UPPER(f.family_category) Then f.family_category
...
```

Reply:
Also, I'm surprised you need to cast it to text 🤔

Reply:
This was lifted from the download query and isn't necessary here. Have removed the title casing validation here.

Reply:
Turns out this is needed - family category is stored in UPPERCASE in the db and the pipeline expects INITCAP - keeping as this for now.